code/modules/events/meteor_wave.dm

```diff
@@ -52,6 +52,7 @@
 /datum/round_event/meteor_wave/tick()
 	if(ISMULTIPLE(activeFor, 3))
 		spawn_meteors(5, wave_type) //meteor list types defined in gamemode/meteor/meteors.dm
+		SSshuttle.shuttle_purchase_requirements_met |= "meteor"

```

Comment:
why is this here?

Reply:
umm isn’t this just when meteor waves are spawned?

Reply:
I mean like it seems work, tried to buy it on private server roundstart and it didn’t work, but after triggering meteor event it worked sooo? Is it just shit code that finds a way to work?

Reply:
why would you have it do this every tick instead of just running it on the start of meteor wave?

Reply:
bc im code illiterate

Reply:
aight so I did this:
if("normal")
  .     wave_type = GLOB.meteors_normal
  .     SSshuttle.shuttle_purchase_requirements_met |= "meteor"
   .     minor_announce("'Asteroid With Engines Strapped To It' Escape Shuttle now available!")

And only did that for normal through catastrophic, excluding all special types of meteor waves. But I need a delay (bc it automatically tells everyone about the shuttle, before an admin announcement (another issue being if admins choose to keep the meteors silent, the new shuttle announcement still gives it away). On top of this, I want it to only make the announcement once, since everytime a meteor wave comes it repeats the shuttle announcement, ideas? I also plan to make an announcement for bubblegum dying, that at least will be more straightforward.

Reply:
just make a list tbh

```DM
    var/static/list/string2meteortype = list(
        "normal" = GLOB.meteors_normal, "threatening" = GLOB.meteors_threatening //etcetc)
    If(!wave_name)
        //don't edit old code it's fine
    wave_type = string2meteortype[wave_type] //die switch die
```
Then in announce
```DM
    if(wave_type in list("normal","threatening","catastrophic") && !("meteor" in SSshuttle.shuttle_purchase_requirements_met))
        SSshuttle.shuttle_purchase_requirements_met |= "meteor"
        minor_announce("'Asteroid With Engines Strapped To It' Escape Shuttle now available!")

Reply:
Second part is good, I added the !meteor and it fixed the repeating. How would I add a delay to the announcement? 

Also do you have any clue my my PR failed the check?

Reply:
the announcement should only run when the other announcement is ran hence why I said to put it in `announce()`

Reply:
The other announce is just a fake announcement tho, should i create another announce proc thingy for it separately?